com.unity.render-pipelines.core/Runtime/Debugging/DebugUI.cs

```diff
@@ -235,11 +244,83 @@ public Value()
             /// Returns the value of the widget.
             /// </summary>
             /// <returns>The value of the widget.</returns>
-            public object GetValue()
+            public virtual object GetValue()
             {
                 Assert.IsNotNull(getter);
                 return getter();
             }
+
+            /// <summary>
+            /// Returns the formatted value string for display purposes.
+            /// </summary>
+            /// <param name="value">Value to be formatted.</param>
+            /// <returns>The formatted value string.</returns>
+            public virtual string FormatString(object value)
+            {
+                return string.IsNullOrEmpty(formatString) ? $"{value}" : string.Format(formatString, value);
```

Comment:
I guess this would be a good place to switch to [MutableString](https://docs.google.com/document/d/1MUWvnECRWz71FymdOtSoJ_Z8cj7-oLcTuijhg-jVfYs) once available...

Reply:
Yes, absolutely. I've spent some time trying to eliminate allocs but this was the one place I couldn't easily fix without removing support for format strings altogether, which I would like to keep. Great to hear that there's work for a non-allocating StringBuilder/String.Format that I could use in the future!